### PR TITLE
device: tx_key caching fix

### PR DESCRIPTION
**Pull request by ph4r05**
- Ignore null keys stored as tx keys as those are invalid. TX key cache can be full of zero keys which blocked tx key recovery using Trezor

Required for https://github.com/monero-project/monero-gui/issues/2183

### CODE REVIEW

File: src/wallet/wallet2.cpp [hunk 1]
```diff
@@ -10355,6 +10364,12 @@ bool wallet2::get_tx_key(const crypto::hash &txid, crypto::secret_key &tx_key, s
   tx_keys.erase(tx_keys.begin());
   additional_tx_keys = tx_keys;
 
+  if (store_tx_info())
+  {
+    m_tx_keys.insert(std::make_pair(txid, tx_key));
+    m_additional_tx_keys.insert(std::make_pair(txid, additional_tx_keys));
+  }
+
```
Comment (moneromooo-monero) [inline src/wallet/wallet2.cpp]: What is the improved UI for that part in practice ? Faster to query next time ?

Comment (ph4r05) [reply src/wallet/wallet2.cpp]: Yes. The tx key recovery requires a round trip with the Trezor, with user confirmation on the device. But I am still thinking about dropping this and let the user do the device recovery again. What do you think?

Comment (moneromooo-monero) [reply src/wallet/wallet2.cpp]: I think it's unlikely the user has to use it repeatedly. Is this wrong ?

Comment (ph4r05) [reply src/wallet/wallet2.cpp]: GUI calls the getTxKey with each transaction info rendering. But if we cache it in the GUI layer it won't 
For security reasons I will drop this store to cache feature.

**Comment by ph4r05** [comment]
@fluffypony do you pls plan to merge this month? :)

**Pull request merged**